graph-proxy/src/graphql/workflows.rs

```diff
@@ -347,4 +347,118 @@ impl WorkflowsQuery {
             .into_result()?;
         Ok(workflow.try_into()?)
     }
+
+    #[instrument(skip(self, ctx))]
+    async fn workflows(
+        &self,
+        ctx: &Context<'_>,
+        proposal_code: String,
+        proposal_number: u32,
+        visit: u32,
+        cursor: Option<String>,
+        limit: Option<u32>,
+    ) -> anyhow::Result<Connection<OpaqueCursor<String>, Workflow, EmptyFields, EmptyFields>> {
+        let server_url = ctx.data_unchecked::<ArgoServerUrl>().deref();
+        let auth_token = ctx.data_unchecked::<Option<Authorization<Bearer>>>();
+        let namespace = format!("{}{}-{}", proposal_code, proposal_number, visit);
+        let client = reqwest::Client::new();
+        let request = |url: reqwest::Url| {
+            if let Some(auth_token) = auth_token {
+                client.get(url).bearer_auth(auth_token.token())
+            } else {
+                client.get(url)
+            }
+        };
+        let mut url = server_url.clone();
+        url.path_segments_mut()
+            .unwrap()
+            .extend(["api", "v1", "workflows", &namespace]);
+        let limit = limit.unwrap_or(10).min(10);
```

Comment:
Could we use an [input value validator instead](https://async-graphql.github.io/async-graphql/en/input_value_validators.html)? Think they automagically provide good error messages